critiquebrainz/ws/review/views.py

```diff
@@ -233,19 +235,36 @@ def review_modify_handler(review_id, user):
     """
 
     def fetch_params():
-        text = Parser.string('json', 'text', min=REVIEW_MIN_LENGTH, max=REVIEW_MAX_LENGTH)
-        return text
+        try:
+            text = Parser.string('json', 'text', min=REVIEW_TEXT_MIN_LENGTH, max=REVIEW_TEXT_MAX_LENGTH)
+        except MissingDataError:
+            text = 'same'                        # Assign temporary value which indicates no modification
+        try:
+            rating = Parser.int('json', 'rating', min=REVIEW_RATING_MIN, max=REVIEW_RATING_MAX)
+        except MissingDataError:
+            rating = 0                           # Assign temporary value which indicates no modification
+        if text is None and rating is None:
+            raise InvalidRequest(desc='Review must have either text or rating')
+        return text, rating
 
     review = get_review_or_404(review_id)
     if review["is_hidden"]:
         raise NotFound("Review has been hidden.")
     if str(review["user_id"]) != user.id:
         raise AccessDenied
-    text = fetch_params()
+    text, rating = fetch_params()
+    if text == 'same':
+        text = review['text']
+    if rating == 0:
+        rating = review['rating']
+    if (text == review['text']) and (rating == review['rating']):
+        raise InvalidRequest(desc='Either text or rating should be edited to update the review')
```

Comment:
I think we discussed this, but this shouldn't be a requirement. Users don't need to think about that.

Reply:
Ok. If none of the parameters are modified, it won't throw an exception and won't make a new revision as well.